src/main/java/com/google/sps/data/TimeRangedText.java

```diff
@@ -85,4 +85,23 @@ public String getText() {
     public String toString() {
         return OPEN_CURLY + START + startTime + DELIMITER + END + endTime + DELIMITER + TEXT + text + CLOSE_CURLY;
     }
+
+    /**
+     * Tests if two TimeRangedText objects are equal
+     * @param other The other object to check equality with this object
+     * @return Whether the input object has the same parameter values as this object
+     */
+    @Override
+    public boolean equals(Object other) {
+        // Tries to cast Object to a TimeRangedText, and returns false if there is a casting error
+        try {
```

Comment:
Instead of cast and catch the exception, we could do
```
if (other == null || !other instanceof of TimeRangedText) {
  return false;
}
...
```

Reply:
Addressed. Now uses `instanceof`.